Makefile

```diff
@@ -34,16 +34,16 @@ define fetch_go_bin_version
 	@mkdir -p $(BIN_DIR)
 
 	@echo ">> fetching $(1)@$(2) revision/version"
-	@if [ ! -d "$(TMP_GOPATH)/src/$(1)" ]; then \
-    	GOPATH=$(TMP_GOPATH) go get -d -u $(1); \
-    else \
-    	git fetch; \
```

Comment:
Hm, why not just `cd $(TMP_GOPATH)/src/$(1) && git fetch` ?

Reply:
Yeah, that could do, given that `$(TMP_GOPATH)/src/$(1)` always expands to a nice value. Meaning absolute, no spaces, no leading dashes.

I'm a CDPATH user and noticed that scripts often break because they're not being prepared for this. So I adopted a more defensive coding style when scripting with `cd` and placeholders in path values.

So a short reasoning about the current structure:

1. `(...)` so that it's a subshell and the `cd` is just confined to that subshell for the call to `git fetch`
2. `CDPATH=''` explicitly disable CDPATH for users who use that (I have e.g. GOPATH in it)
3. `--` explicitly end the flag phase, just in case someone adds a dash in front of TMP_GOPATH
4. single quotes around the path: Expansion is done by make, not by shell, and we don't want another shell expansion to happen

We shouldn't leave out `1.`, but can leave out `3.` if the path never starts with dashes, `2.` if it's always absolute (implicitly includes `3.`) and `4.` if the path will never contain spaces or other special characters interpreted by shells.

I admit that some of the precautions deal with unlikely cases, but I think it's worth doing it. WDYT?

Reply:
Cool, I am fine with doing this but would be nice to either do nowhere those, or do them everywhere here in this function, right? 

You have similar `cd` below this line.

Reply:
Wow :see_no_evil: I'm blind ^^

Reply:
I think the subshell isn't necessary, since I guess that make will always spawn a new shell for each command...

Reply:
Done.